src/jit/codegenarm64.cpp

```diff
@@ -450,6 +478,56 @@ void CodeGen::genBuildRegPairsStack(regMaskTP regsMask, ArrayStack<RegPair>* reg
         }
     }
     assert(regsCount == 0 && regsMask == RBM_NONE);
+
+    genSetUseSaveNextPairs(regStack);
+}
+
+//------------------------------------------------------------------------
+// genSetUseSaveNextPairs: Set useSaveNextPair for each RegPair on the stack which unwind info can be encoded as
+// save_next code.
+//
+// Arguments:
+//   regStack - a regStack instance to set useSaveNextPair.
+//
+// Notes:
+// We can use save_next for RegPair(N, N+1) only when we have sequence like (N-2, N-1), (N, N+1).
+// In this case in the prolog save_next for (N, N+1) refers to save_pair(N-2, N-1);
+// in the epilog the unwinder will search for the first save_pair (N-2, N-1)
+// and then go back to the first save_next (N, N+1) to restore it first.
+//
+// static
+void CodeGen::genSetUseSaveNextPairs(ArrayStack<RegPair>* regStack)
+{
+    for (int i = 1; i < regStack->Height(); ++i)
+    {
+        RegPair& curr = regStack->BottomRef(i);
+        RegPair  prev = regStack->Bottom(i - 1);
+
+        if (prev.reg2 == REG_NA || curr.reg2 == REG_NA)
+        {
+            continue;
+        }
+
+        if (genCanUseSaveNextPair(prev, curr))
+        {
+            curr.useSaveNextPair = true;
+        }
+    }
+}
+
+//------------------------------------------------------------------------
+// genCanUseSaveNextPair: Check that the curr regPair is adjusted to the next.
+//
+// Arguments:
+//   curr - a RegPair with 2 regs that we want to check;
+//   next - the next RegPair from the stack, also have to have 2 regs.
+//
+// static
+bool CodeGen::genCanUseSaveNextPair(RegPair curr, RegPair next)
+{
+    assert(REG_NEXT(curr.reg1) == curr.reg2);
+    assert(REG_NEXT(next.reg1) == next.reg2);
+    return (REG_NEXT(curr.reg2) == next.reg1);
```

Comment:
I'm thinking that the case of the first FP reg pair being `save_next` is so rare, and the potential benefit so tiny, that we shouldn't even support it. (We'd have zero or close to zero coverage of this case, too.) That would probably simplify the code slightly, too.

Reply:
> That would probably simplify the code slightly, too.

The common handling for int and float masks simplies the code, https://github.com/dotnet/coreclr/pull/21683/commits/b6ca00fdd435f41f803046614bb551ea3010f4c3 shows +33 −82 lines diff.

Reply:
I think we can keep the common handling but do not use `save_next` for the first float pair.

Espessialy because `unwinder/arm64/unwinder_arm64.cpp` does not support it (or I do not see how it supports it):
https://github.com/dotnet/coreclr/blob/4f704d172863b50492a4a4254691fe1374a55b62/src/unwinder/arm64/unwinder_arm64.cpp#L938-L950

where `RtlpUnwindRestoreRegisterRange`  "Restores a series of integer registers from the stack" and doesn't expect float registers in `RegisterCount`.